ansible/roles/test/files/ptftests/advanced-reboot.py

```diff
@@ -787,6 +786,56 @@ def pre_reboot_test_setup(self):
         watcher = self.pool.apply_async(self.reachability_watcher)
         time.sleep(5)
 
+    def get_warmboot_finalizer_state(self):
+        stdout, stderr, _ = self.dut_connection.execCommand('sudo systemctl is-active warmboot-finalizer.service')
+
+        if stdout == []:
+            self.log('Finalizer state not returned from DUT')
+            #self.fails['dut'].add('Error collecting Finalizer state from DUT')
+            return ''
+        if stderr != []:
+            self.fails['dut'].add("stderr from DUT while collecting Finalizer state: %s" % (str(stderr)))
+
+        finalizer_state = stdout[0].strip()
+        return finalizer_state
+
+    def get_now_time(self):
+        stdout, stderr, _ = self.dut_connection.execCommand('date +"%Y-%m-%d %H:%M:%S"')
+        if stdout == []:
+            self.fails['dut'].add('Error collecting current date from DUT: empty value returned')
+        if stderr != []:
+            self.fails['dut'].add("Error collecting current date from DUT: %s" % (str(stderr)))
+        return datetime.datetime.strptime(stdout[0].strip(), "%Y-%m-%d %H:%M:%S")
```

Comment:
this will raise an error if stdout is empty

Reply:
Right. Added the error handling by raising Exception.